aten/src/ATen/native/cuda/MaxUnpooling.cu

```diff
@@ -41,11 +41,13 @@ __global__ void max_unpooling2d_forward_kernel(
     const int64_t outputHeight,
     const int64_t outputWidth,
     T* output) {
+  int64_t outputImageSize = outputHeight * outputWidth;
   CUDA_KERNEL_LOOP(linearIndex, numInputElements) {
     int c = (linearIndex / inputWidth / inputHeight) % numChannels;
     int n = linearIndex / inputWidth / inputHeight / numChannels;
     output += (n * numChannels + c) * outputHeight * outputWidth;
     int maxind = indices[linearIndex];
+    CUDA_KERNEL_ASSERT(maxind >= 0 && maxind < outputImageSize);
```

Comment:
This could probably lead to perf regression similar to https://github.com/pytorch/pytorch/issues/74751

wdyt @ngimel?